### DISCUSSION TITLE
New slot in sidebar

### DISCUSSION THREAD

**Discussion by SebastianBienert** [scalar/scalar#909] [Ideas]
Starting from recent change: https://github.com/scalar/scalar/pull/857 we can use `SidebarGroup` and `SidebarElement` in internal implementation. If someone would like to expand `Sidebar` with custom elements it has to be placed in either `<sidebar-start>` or `<sidebar-end>` slot. Both slots are out of scroll section in [sidebar](https://github.com/scalar/scalar/blob/6da73708130065bb75d17dbb4d04b9cb351123b0/packages/api-reference/src/components/Sidebar.vue#L57C1-L58C1). I suggest making additional slots inside of scrollable [div](https://github.com/scalar/scalar/blob/6da73708130065bb75d17dbb4d04b9cb351123b0/packages/api-reference/src/components/Sidebar.vue#L58) (eg. `sidebar-content-start`, `sidebar-content-end`) which would allow to expand main content of the sidebar while still having one scrollable section.

<img width="183" alt="image" src="https://github.com/scalar/scalar/assets/17458785/14ee1fa8-2fda-487e-b7cb-7325c9582a52">

**Comment by marclave** [comment]
@hwkr thoughts on this? :)

**Reply by hwkr** [reply]
Seems like a good idea to me! Shouldn't be too hard to add I don't think
